pkg/converter/configuration/converter.go

```diff
@@ -0,0 +1,177 @@
+// limitations under the License.
+
+// Package configuration has generic functions to get the new provider names
+// from compositions and managed resources.
+package configuration
+
+import (
+	"fmt"
+	"strings"
+
+	xpmetav1 "github.com/crossplane/crossplane/apis/pkg/meta/v1"
+	xpmetav1alpha1 "github.com/crossplane/crossplane/apis/pkg/meta/v1alpha1"
+	xppkgv1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	xppkgv1beta1 "github.com/crossplane/crossplane/apis/pkg/v1beta1"
+	"github.com/pkg/errors"
+	"github.com/upbound/upjet/pkg/migration"
+)
+
+const (
+	awsPackage   = "xpkg.upbound.io/upbound/provider-aws"
+	azurePackage = "xpkg.upbound.io/upbound/provider-azure"
+	gcpPackage   = "xpkg.upbound.io/upbound/provider-gcp"
+)
+
+type mRPreProcessor struct {
+	ProviderNames map[string]struct{}
+}
+
+func NewMRPreProcessor() *mRPreProcessor {
+	return &mRPreProcessor{
+		ProviderNames: map[string]struct{}{},
+	}
+}
+
+type compositionPreProcessor struct {
+	ProviderNames map[string]struct{}
+}
+
+func NewCompositionPreProcessor() *compositionPreProcessor {
+	return &compositionPreProcessor{
+		ProviderNames: map[string]struct{}{},
+	}
+}
+
+// GetSSOPNameFromManagedResource collects the new provider name from MR
+func (mp *mRPreProcessor) GetSSOPNameFromManagedResource(u migration.UnstructuredWithMetadata) error {
+	for _, pn := range getProviderAndServiceName(u.Object.GroupVersionKind().Group) {
+		mp.ProviderNames[pn] = struct{}{}
+	}
+	return nil
+}
+
+// GetSSOPNameFromComposition collects the new provider name from Composition
+func (cp *compositionPreProcessor) GetSSOPNameFromComposition(u migration.UnstructuredWithMetadata) error {
+	composition, err := migration.ToComposition(u.Object)
+	if err != nil {
+		return errors.Wrap(err, "unstructured object cannot be converted to composition")
+	}
+	for _, composedTemplate := range composition.Spec.Resources {
+		composedUnstructured, err := migration.FromRawExtension(composedTemplate.Base)
+		if err != nil {
+			return errors.Wrap(err, "resource raw cannot convert to unstructured")
+		}
+		for _, pn := range getProviderAndServiceName(composedUnstructured.GroupVersionKind().Group) {
+			cp.ProviderNames[pn] = struct{}{}
+		}
+	}
+	return nil
+}
+
+func getProviderAndServiceName(name string) []string {
+	parts := strings.Split(name, ".")
+	if len(parts) > 3 {
+		provider := ""
+		switch parts[1] {
+		case "aws":
+			provider = "provider-aws"
+		case "gcp":
+			provider = "provider-gcp"
+		case "azure":
+			provider = "provider-azure"
+		default:
+			return nil
+		}
+		service := parts[0]
+		return []string{fmt.Sprintf("%s-%s", provider, service), fmt.Sprintf("provider-family-%s", parts[1])}
+	}
+	return nil
+}
+
+type ConfigMetaParameters struct {
+	familyVersion        string
+	monolith             string
+	compositionProcessor *compositionPreProcessor
+}
+
+type ConfigPkgParameters struct {
+	regorg        string
+	familyVersion string
```

Comment:
If I'm not mistaken, this is the tag to be used for the migration target configuration package and has no relation to any provider family's version:
```suggestion
	packageVersion string
```